selfapi.js

```diff
@@ -367,6 +374,24 @@ API.prototype = {
     });
   },
 
+  // Build index routes.
+  buildIndexRoutes: function (basePath) {
+      var fullPath = normalizePath(this.path, basePath) || '/';
+      this.get({
+          title: 'Index',
+          handler: (request, response) => {
+              var routes = {};
+              Object.keys(this.children).forEach(child => {
+                  if (child == "null") {
+                      return;
+                  }
+                  routes[child.replace(/^\//, '')] = nodepath.join(fullPath, child);
```

Comment:
Ideally, there should be absolute URLs, see [what Github does for instance](https://api.github.com/). This code is a quick and dirty PoC, I should review it.

Do you know if there is a way to get the absolute URL?